src/Microsoft.TemplateEngine.Abstractions/ISettingsLoader.cs

```diff
@@ -28,7 +28,7 @@ public interface ISettingsLoader
 
         bool TryGetMountPointInfo(Guid mountPointId, out MountPointInfo info);
 
-        void WriteTemplateCache(IList<ITemplateInfo> templates, string locale, bool isCurrentCache);
+        void WriteTemplateCache(IList<ITemplateInfo> templates, string cacheVersion, string locale, bool isCurrentCache);
```

Comment:
Shouldn't the version of the cache the engine requires be an implementation detail of the settings loader or template cache and not need to be exposed?